pgdrive/scene_creator/highway_vehicle/kinematics.py

```diff
@@ -31,18 +32,20 @@ class Vehicle:
     """ Maximum reachable speed [m/s] """
     def __init__(
         self,
-        scene: SceneManager,
+        traffic_mgr: TrafficManager,
         position: List,
         heading: float = 0,
         speed: float = 0,
         np_random: np.random.RandomState = None,
     ):
-        self.scene = scene
+        self.traffic_mgr = traffic_mgr
         self._position = np.array(position).astype('float')
         self.heading = heading
         self.speed = speed
-        self.lane_index, _ = self.scene.network.get_closest_lane_index(self.position) if self.scene else np.nan
-        self.lane = self.scene.network.get_lane(self.lane_index) if self.scene else None
+        self.lane_index, _ = self.traffic_mgr.map.road_network.get_closest_lane_index(
+            self.position
+        ) if self.traffic_mgr else np.nan
```

Comment:
How you assign a nan to two value? This is buggy.

Reply:
got it